src/devices/PiJuice/PiJuiceStatus.cs

```diff
@@ -132,13 +132,7 @@ public ElectricCurrent GetBatteryCurrent()
         {
             var response = _piJuice.ReadCommand(PiJuiceCommand.BatteryCurrent, 2);
 
-            int i = BinaryPrimitives.ReadInt16LittleEndian(response);
-            if ((i & (1 << 15)) == (1 << 15))
-            {
-                i -= (1 << 16);
-            }
-
-            return new ElectricCurrent(i, ElectricCurrentUnit.Milliampere);
+            return new ElectricCurrent(BinaryPrimitives.ReadInt16LittleEndian(response), ElectricCurrentUnit.Milliampere);
```

Comment:
this looks good but consider using `ElectricCurrent.FromMilliampere(BinaryPrimitives.ReadInt16LittleEndian(response))`

Reply:
Hi @krwq, I've changed all occurrences of UnitsNet constructors to the static From methods in the code.